### ISSUE TITLE
iOS - Implement Rotate Image for Take/Pick and update metadata.

### ISSUE THREAD

**Issue by jamesmontemagno** [jamesmontemagno/MediaPlugin#334]
This is via #333

**Comment by jamesmontemagno** [comment]
Also via: https://github.com/jamesmontemagno/MediaPlugin/issues/304

**Comment by weareformative** [comment]
Do you have an ETA for the implementation of this feature? Are there any workarounds that you can point us to in the meantime?

**Comment by jamesmontemagno** [comment]
You could use FFImageLoading, which will respect the metadata exif info. That should work fine.

**Comment by weareformative** [comment]
Thank you James, will do! Again, thanks again for your quick responses.

**Comment by bordline23** [comment]
Hi @jamesmontemagno,
I've been confused about something when using the plugin (issue is in iOS only) and upload the image.
1.  Upload the image from PickPhotoAsync, it uploaded as landscape.
2.  Upload the image from TakePhotoAsync, it uploaded as portrait (Just fine)
3. Use ImageResizer in my API to Autorotate the image, #1 would now be portrait and #2 would also be rotated (to the right).
4. Use it on FFImage and assign the source to ImageSource.FromFile(Pick/TakePhoteAsync's file path) it displays the same way in no. 3.

**Comment by weareformative** [comment]
Hello @bordline23,

What worked for me was removing AllowCropping = true from the options configuration. This option was causing all the rotation issues I was encountering in my app with iOS. FFImageLoading alone did not solve the issue for me. I hope that helps you. 

Thank you,
Andrew

**Comment by tacoman667** [comment]
@weareformative But, what does `AllowCropping` do? How does it affect the rest of the functionality?

**Comment by bordline23** [comment]
@weareformative, that worked! Thanks.

I haven't gone deep into it but I think the cropping makes the image still while the EXIF metadata remains the same, e.g. for portrait its turned counter clockwise, that's why its misleading.

**Comment by jamesmontemagno** [comment]
@tacoman667 If you set "AllowCropping" = true then it brings up the cropping tool after you take a photo. It most likely is that the cropping might not rotate correctly.

**Comment by jamesmontemagno** [comment]
It is merge from https://github.com/jamesmontemagno/MediaPlugin/issues/304 into this issue as well.

**Comment by benevbright** [comment]
I also tried 'AllowCropping = true'. And I got rotated photo from took a photo, which is wrong.

**Comment by tacoman667** [comment]
PR has been submitted.

**Comment by weareformative** [comment]
@bordline23, @tacoman667, @benevbright, & @jamesmontemagno

UPDATE: Removing 'AllowCropping = true' worked for iOS in so much that the image rotation was correctly displayed in my CachedImage view. However, that caused the image to be uploaded incorrectly rotated to the server. Therefore, when viewing the image on an Android device, or on the server, the image was rotated incorrectly because that is how the actual file was saved.

The image displays correctly in both iOS and Android if you set 'AllowCropping = true' and set the Image or CachedImage view's Source property to the URL (ImageSource.FromUri) where the photo was uploaded to, assuming that you're uploading the photo to a remote server. 

In summary, the rotating image problem appears when you set the image view's Source property FromFile or FromStream. However, AllowCropping should be set to true for the actual image file to be saved in the correct orientation.

I hope that helps.

**Comment by dlpopescumtk** [comment]
@jamesmontemagno : I'm getting _java.io.IOException: Invalid marker: 89_ in the VS output when running the Android project, using the FFImageLoading library. Could this be related to not using the Xamarin.Support.Exif in the library? Apparently, the old ExifInterface gets initialized for PNG files as well, which do not embed exif info.
